dcm4che-imageio/src/main/java/org/dcm4che3/imageio/stream/BulkURIImageInputStream.java

```diff
@@ -0,0 +1,84 @@
+//
+///////////////////////////////////////////////////////////////
+//                C O P Y R I G H T  (c) 2019                //
+//        Agfa HealthCare N.V. and/or its affiliates         //
+//                    All Rights Reserved                    //
+///////////////////////////////////////////////////////////////
+//                                                           //
+//       THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF      //
+//        Agfa HealthCare N.V. and/or its affiliates.        //
+//      The copyright notice above does not evidence any     //
+//     actual or intended publication of such source code.   //
+//                                                           //
+///////////////////////////////////////////////////////////////
+//
+package org.dcm4che3.imageio.stream;
+
+import javax.imageio.stream.ImageInputStream;
+import javax.imageio.stream.ImageInputStreamImpl;
+import java.io.IOException;
+
+/**
+ * Return a subset of an image input stream.  The local variables track the location within the offset/length
+ * bounds which are mapped onto the underlying full image input stream.
+ * @author Andrew Cowan (awcowan@gmail.com)
+ */
+public class BulkURIImageInputStream extends ImageInputStreamImpl {
```

Comment:
Really simple segmented IIS implementation.

The current SegmentedImageInputStream has a lot of knowledge about how BulkData fragments are stored, which is fine, but I would like to be able to generically point to a subset of a URI and this class provides that.